client/src/documentLinkProvider.ts

```diff
@@ -27,20 +27,49 @@ export class BitbakeDocumentLinkProvider implements vscode.DocumentLinkProvider
-    const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+    const parentDir = uri.path.split('/').slice(0, -1).join('/')
+    const pnDir = path.join(parentDir, extractRecipeName(uri.path) as string)
     const filesDir = path.join(parentDir, 'files')
-    return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
-      ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    try {
+      return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
+        ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    } catch (err) {
+      logger.error(`An error occurred when finding files with pattern: ${filenamesRegex}. ${JSON.stringify(err)}`)
+      return []
+    }
+  }
+
+  private findFolders (links: Awaited<RequestResult['getLinksInDocument']>): Awaited<RequestResult['getLinksInDocument']> {
+    return links.filter(link => path.parse(link.value).ext === '')
   }
 
   async provideDocumentLinks (document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.DocumentLink[]> {
     const linksData = await this.client.sendRequest<RequestResult['getLinksInDocument']>(RequestMethod.getLinksInDocument, { documentUri: document.uri.toString() })
     const documentLinks: vscode.DocumentLink[] = []
-    const relatedFiles = await this.findRelatedFiles(document, linksData, token)
-    logger.debug(`Found ${relatedFiles.length} local SRC_URI files for ${document.uri.toString()}`)
+    const relatedFiles = await this.findRelatedFiles(document.uri, linksData, token)
+    const folders = this.findFolders(linksData)
+
+    logger.debug(`Found ${folders.length} folders from the links for file: ${document.uri.toString()}`)
+    logger.debug(`Found ${relatedFiles.length} files from the links for file: ${document.uri.toString()}`)
+
+    // Reveal folders in explorer
+    folders.forEach((folder) => {
+      const parentDir = document.uri.path.split('/').slice(0, -1).join('/')
+      const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+      const commandArguments = [path.join(pnDir, folder.value)]
+      const targetUri = vscode.Uri.parse(`command:bitbake.revealInExplorer?${encodeURIComponent(JSON.stringify(commandArguments))}`)
+      /*
+        targetUri.path === targetUri.fsPath === "bitbake.revealInExplorer",
+        targetUri.scheme === "command",
+        targetUri.query is the array of arguments
+
+        Assigning command uris to document links will trigger the command on ctrl+click with the arguments
+      */
+      documentLinks.push(new vscode.DocumentLink(folder.range, targetUri))
```

Comment:
The feature seem to treat any link that doesn't have a file extension as a folder now. In poky/meta/recipes-core/busybox/busybox_1.36.1.bb for instance, `file://mdev` doesn't open anything now.

Reply:
yes, the approach to get the folders needs to improve. The follow-link not working on some links is mentioned in another conversation.